axiom/nr_txn.c

```diff
@@ -157,6 +173,32 @@ void nr_txn_set_long_attribute(nrtxn_t* txn,
                                attribute->name, value);
 }
 
+void nr_txn_attributes_set_string_attribute(nr_attributes_t* attributes,
+                                            const nr_txn_attribute_t* attribute,
+                                            const char* value) {
+  if (NULL == attribute) {
+    return;
+  }
+  if (NULL == value) {
+    return;
+  }
+  if ('\0' == value[0]) {
+    return;
+  }
+  nr_attributes_agent_add_string(attributes, attribute->destinations,
+                                 attribute->name, value);
+}
+
+void nr_txn_attributes_set_long_attribute(nr_attributes_t* attributes,
+                                          const nr_txn_attribute_t* attribute,
+                                          long value) {
```

Comment:
Would it make sense for this to be `const long value`?

Reply:
Please see: [921224e](https://github.com/newrelic/newrelic-php-agent/pull/585/commits/921224e3c10ba59cc9249787ad25d1dcdf519d2a)